openedge-plugin/src/main/java/org/sonar/plugins/openedge/foundation/OpenEdgeSettings.java

```diff
@@ -249,13 +280,30 @@ private final void parseBuildDirectory() {
         System.currentTimeMillis() - currTime, numClasses.get(), numMethods.get(), numProperties.get());
   }
 
+  private TypeInfo parseRCode(File file) {
+    try (FileInputStream fis = new FileInputStream(file)) {
+      LOG.debug("Parsing rcode {}", file.getAbsolutePath());
+      RCodeInfo rci = new RCodeInfo(fis);
+      if (rci.isClass()) {
+        return rci.getTypeInfo();
+      }
+    } catch (InvalidRCodeException | IOException | RuntimeException caught) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1166)